### PR TITLE
Show a change when updating case properties to blank values 

### PR DESCRIPTION
**Pull request by proteusvacuum**
https://manage.dimagi.com/default.asp?275838

And other miscellaneous bugs I found along the way.

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_get_case_property_changes.py [hunk 1]
```diff
@@ -9,17 +10,18 @@
...
 )
 from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
 from corehq.form_processor.tests.utils import run_with_all_backends
 
 
 class TestCasePropertyChanged(TestCase):
     def setUp(self):
-        self.factory = CaseFactory('domain')
-        self.case = self.factory.create_case(owner_id='owner')
+        self.domain = "isildur"
```
Comment (jmtroth0) [inline corehq/ex-submodules/casexml/apps/case/tests/test_get_case_property_changes.py]: ❤️

File: corehq/ex-submodules/casexml/apps/case/util.py [hunk 2]
```diff
@@ -177,6 +178,12 @@ def property_changed_in_action(case_transaction, case_id, case_property_name):
     from casexml.apps.case.xform import get_case_updates
     PropertyChangedInfo = namedtuple("PropertyChangedInfo", 'transaction new_value modified_on')
     include_create_fields = case_property_name in ['owner_id', 'name', 'external_id']
+
+    if not should_use_sql_backend(case_transaction.form.domain):
+        # couch domains return 2 transactions for case properties created in a create form
+        if case_transaction.is_case_create and not include_create_fields:
```
Comment (jmtroth0) [inline corehq/ex-submodules/casexml/apps/case/util.py]: By the names, it seems like this is something that shouldn't happen? Is this comment saying that for couch domains another transaction will add these fields?

**Review by jmtroth0 — APPROVED** [review]
1 question, but generally lgtm

File: corehq/ex-submodules/casexml/apps/case/util.py (hunk 2 above)
Comment (proteusvacuum) [reply corehq/ex-submodules/casexml/apps/case/util.py]: yeah, this is something that surprised me, and seems to be inconsistent between SQL and Couch. 

In SQL, when you create a case with a bunch of case properties you only get a single `CaseTransaction` - A "create" transaction, which contains a `create` action (which creates things like the name and owner_id) and an `update` action (which creates the case properties).

In couch, when you create a case with a bunch of case properties, you get 2 `CommCareCaseAction`s - A "create" which has both a `create` and `update` actions, and an "update" which only has an `update`.

**Pull request merged**